network/peer/network.go

```diff
@@ -35,6 +36,8 @@ type Network interface {
 	// Peers returns peers that are not known.
 	Peers(
 		peerID ids.NodeID,
+		trackedSubnets set.Set[ids.ID],
+		requestAllPeers bool,
```

Comment:
This is really the whole point of the PR. This allows peers to learn about non-primary network validators.